proxmox/ssh/client.go

```diff
@@ -332,6 +356,17 @@ func (c *client) NodeStreamUpload(
 		return err
 	}
 
+	if d.Mode != "" {
+		parsedFileMode, parseErr := strconv.ParseUint(d.Mode, 8, 12)
+		if parseErr != nil {
+			return fmt.Errorf("failed to parse file mode %q: %w", d.Mode, err)
+		}
+
+		if err = c.changeModeUploadedFile(ctx, sshClient, remoteFilePath, os.FileMode(uint32(parsedFileMode))); err != nil {
```

Comment:
This would work only for `root@pam` auth. For a non-root user we have to run `try_sudo chmod ...`, similar to what we're doing in `uploadFile` func.

On the other hand, you need to have root permissions to set up a hook script in the first place. So, I think it is probably OK to have this limitation for now.

Also, while it's great to have this heature also [captured in examples](https://github.com/bpg/terraform-provider-proxmox/pull/1478/files#diff-a919798b982e82917e26802715bf1dac45c104536505c6b57a5354d49944a842R76), I can't run my automated test suite with the mode set :( 
So could you pls. rather move it to the [`virtual_environment_file.md`](https://github.com/bpg/terraform-provider-proxmox/blob/2e34c57f6c95bff2f3e35ee617e783a78616c900/docs/resources/virtual_environment_file.md?plain=1#L46) doc?

Reply:
Regarding the `root@pam` requirement I could rewrite it to use `try_sudo`. Even though the `proxmox_virtual_environment_file` resource is used mainly for snippets (because `proxmox_virtual_environment_download_file` should be preferred for everything else), IÄm not sure if we should mix it too tightly with the use of the VM/CT `hook_script_file_id` attribute. Is it always required to use `root@pam` to interact with `snippets` enabled stores? If so I guess documenting is the only way, otherwise I'd adjust the implementation to reduce the use of `root@pam`.

Regarding the example: I'll move it into a dedicated section in the docs 👍🏼

Reply:
> Is it always required to use root@pam to interact with snippets enabled stores?

Technically, yes, since the snippets folder is writable only by root@pam. The `try_sudo` and `tee` command [via sudo](https://github.com/bpg/terraform-provider-proxmox/blob/main/docs/index.md?plain=1#L202) work around this.

However, I'm not in favour of adding chmod to sudoers permissions, as it poses a greater security risk IMO.

So, I'd suggest documenting the root@pam requirement for this PR.

Reply:
Actually, there is a related ticket that I haven't look at yet:
- https://github.com/bpg/terraform-provider-proxmox/issues/1475

Reply:
I absolutely agree regarding the risk of granting too much `sudoers` permissions for `chmod`, even though access to storage locations should be limited and audited.

Even when we implement it without `root@pam` in this PR, with #1475 the need to use `root@pam` might be necessary anyway when the connected storage has the same limitations. And the implementation can be adjusted if #1475 can be solved without `root@pam` so I'll simply adjust the docs to get this PR done.